### PR TITLE
test: coverage for duplicated key err

### PR DESCRIPTION
**Pull request by saeidee**
- [X] Do only one thing
- [X] Non breaking API changes
- [X] Tested

### What did this pull request do?

Test coverage for ErrDuplicatedKey, related issue: https://github.com/go-gorm/gorm/issues/6345

### CODE REVIEW

**Comment by saeidee** [comment]
Seems like the tests were failing for sqlserver, I will fix that.

**Comment by saeidee** [comment]
@a631807682 here is the fix PR for sqlserver driver https://github.com/go-gorm/sqlserver/pull/103, shall we move with this PR?

File: tests/error_translator_test.go [hunk 1]
```diff
@@ -27,3 +27,34 @@ func TestDialectorWithErrorTranslatorSupport(t *testing.T) {
...
+		gorm.Model
+		Name string `gorm:"unique"`
+	}
+
+	db, err := OpenTestConnection(&gorm.Config{TranslateError: true})
+	if err != nil {
+		t.Fatalf("failed to connect database, got error %v", err)
+	}
+
+	supportedDialectors := map[string]bool{"sqlite": true, "postgres": true, "mysql": true}
```
Comment (a631807682) [inline tests/error_translator_test.go:42]: > @a631807682 here is the fix PR for sqlserver driver [go-gorm/sqlserver#103](https://github.com/go-gorm/sqlserver/pull/103), shall we move with this PR?

Is this no longer needed when https://github.com/go-gorm/sqlserver/pull/103 is merged? If so, I expect to wait for https://github.com/go-gorm/sqlserver/pull/103 to be merged

**Review by a631807682 — APPROVED** [review]

**Comment by ncruces** [comment]
@saeidee can you please take a look at https://github.com/go-gorm/gorm/issues/6435?
I don't know how best to fix this.
Thanks!

**Pull request merged**